includes/User.php

```diff
@@ -671,7 +671,7 @@ public static function isCreatableName( $name ) {
 
 		// Preg yells if you try to give it an empty string
 		if( $wgInvalidUsernameCharacters !== '' ) {
-			if( preg_match( '/[' . preg_quote( $wgInvalidUsernameCharacters, '/' ) . ']/', $name ) ) {
+			if( preg_match( $wgInvalidUsernameCharacters, $name ) ) {
```

Comment:
Not sure if we should change it or keep it for backward compatibility with Mediawiki.
Alternative solution is to use ```AbortNewAccount``` hook.